Mathlib/Tactic/Linarith/Preprocessing.lean

```diff
@@ -192,51 +192,33 @@ end natToInt
 
 section strengthenStrictInt
 
-/--
-`isStrictIntComparison tp` is true iff `tp` is a strict inequality between integers
-or the negation of a weak inequality between integers.
--/
-def isStrictIntComparison (e : Expr) : MetaM Bool := do
-  match (← instantiateMVars e).getAppFnArgs with
-  | (``LT.lt, #[.const ``Int [], _, _, _]) => return true
-  | (``GT.gt, #[.const ``Int [], _, _, _]) => return true
-  | (``Not, #[e]) => match e.getAppFnArgs with
-    | (``LE.le, #[.const ``Int [], _, _, _]) => return true
-    | (``GE.ge, #[.const ``Int [], _, _, _]) => return true
-    | _ => return false
-  | _ => return false
-
 /--
 If `pf` is a proof of a strict inequality `(a : ℤ) < b`,
 `mkNonstrictIntProof pf` returns a proof of `a + 1 ≤ b`,
 and similarly if `pf` proves a negated weak inequality.
+Otherwise, returns `pf` unchanged.
 -/
 def mkNonstrictIntProof (pf : Expr) : MetaM Expr := do
   match (← instantiateMVars (← inferType pf)).getAppFnArgs with
-  | (``LT.lt, #[_, _, a, b]) =>
+  | (``LT.lt, #[.const ``Int [], _, a, b]) =>
     return mkApp (← mkAppM ``Iff.mpr #[← mkAppOptM ``Int.add_one_le_iff #[a, b]]) pf
-  | (``GT.gt, #[_, _, a, b]) =>
+  | (``GT.gt, #[.const ``Int [], _, a, b]) =>
     return mkApp (← mkAppM ``Iff.mpr #[← mkAppOptM ``Int.add_one_le_iff #[b, a]]) pf
   | (``Not, #[P]) => match P.getAppFnArgs with
-    | (``LE.le, #[_, _, a, b]) =>
+    | (``LE.le, #[.const ``Int [], _, a, b]) =>
       return mkApp (← mkAppM ``Iff.mpr #[← mkAppOptM ``Int.add_one_le_iff #[b, a]])
         (← mkAppM ``lt_of_not_ge #[pf])
-    | (``GE.ge, #[_, _, a, b]) =>
+    | (``GE.ge, #[.const ``Int [], _, a, b]) =>
       return mkApp (← mkAppM ``Iff.mpr #[← mkAppOptM ``Int.add_one_le_iff #[a, b]])
         (← mkAppM ``lt_of_not_ge #[pf])
-    | _ => throwError "mkNonstrictIntProof failed: proof is not an inequality"
-  | _ => throwError "mkNonstrictIntProof failed: proof is not an inequality"
-
+    | _ => return pf
+  | _ => return pf
```

Comment:
What you did looks correct in that last transformation, but I think we should back it out to right before that last commit. The `MetaM (Option Expr)` signature gives strictly more information to the caller, which could perhaps be useful one day.

What you can do in `strengthenStrictInt` to keep it similarly short is `return [(← mkNonstrictIntProof h).getD h]`

Reply:
done